### PR TITLE
fixes for websocket assertion failed: 0 (src/protocols/pubsub/xpub.c:138) when closing connection from a browser. #443

### PR DESCRIPTION
**Pull request by zakharov**

### CODE REVIEW

File: src/transports/ws/sws.c [hunk 1]
```diff
@@ -1361,16 +1361,22 @@ static void nn_sws_handler (struct nn_fsm *self, int src, int type,
                     if (sws->inmsg_current_chunk_len == 0)
                     {
                         if (sws->is_final_frame) {
-                            sws->instate = (sws->is_control_frame ?
-                                NN_SWS_INSTATE_RECVD_CONTROL :
-                                NN_SWS_INSTATE_RECVD_CHUNKED);
-                            nn_pipebase_received (&sws->pipebase);
-                            return;
+                           if (sws->opcode ==  NN_WS_OPCODE_CLOSE) {
```
Comment (gdamore) [inline src/transports/ws/sws.c]: nit: there's an extra space here after the ==

**Comment by gdamore** [comment]
@wirebirdlabs you have reviewed this as well, right?  

@zakharov can you rebase this down to a single commit please?  (Please address the one whitespace nit I found while at it.)  Also, the preferred way is to have a bug report and reference it as a single line commit message of the form "fixes #<bugid> <bug synopsis>"

Thanks.

**Comment by wirebirdlabs** [comment]
I can do some testing early this week, but if it works for @zakharov, it
also looks good. My only consideration that i will try to investigate is to
ensure no resources are leaked, and that a connection is entirely cleaned
up under this condition.

-JRD
Sent from a mobile device

On Jul 26, 2015, at 4:30 PM, gdamore <notifications@github.com> wrote:

@wirebirdlabs <https://github.com/wirebirdlabs> you have reviewed this as
well, right?

@zakharov <https://github.com/zakharov> can you rebase this down to a
single commit please? (Please address the one whitespace nit I found while
at it.) Also, the preferred way is to have a bug report and reference it as
a single line commit message of the form "fixes # "

Thanks.

—
Reply to this email directly or view it on GitHub
<https://github.com/nanomsg/nanomsg/pull/447#issuecomment-125042084>.

**Comment by gdamore** [comment]
A new PR was submitted in lieu of this one (#448) closing.

**Pull request closed without merging**